### ISSUE TITLE
Unable to close chrome at the end of a run

### ISSUE THREAD

**Issue by sjieg** [spriteCloud/lapis-lazuli#40]
This issue is to keep track of our research on this ticket:

Fails:
- When running a scenario that is not the last one of a feature (Cucumber 2.0 only)
  
  > This is because of ast.rb:107 when the scenario returns the last scenario of a file and not the last one of the current run
- When running with close_after_scenario("end")
  
  > It looks like the browser connection is closed before the browser has been closed
  > To investigate if this has to do with the cucumber version as well.

**Comment by sjieg** [comment]
Dit more research on the first failure.
`scenario.feature.feature_elements.last` is returning the last scenario of a feature, disregarding if that last scenario is included in the current `@tag` that is running.

I couldn't find any clear documentation of the contents of the scenario object. But below I will add a Yaml export of the object,

The difference between a scenario that is included in the current run and one that is not, is that the scenario included has an hash value "all_locations", and the other one does not. But `p scenario.feature.feature_elements.last.all_locations.nil?` always returns true.

So until now I have note been able to distinguish ignored scenario's of actually running scenario's within a feature.

``` yml
test_steps:
- !ruby/object:Cucumber::Core::Test::Step
  source:
  - &15 !ruby/object:Cucumber::Core::Ast::Feature
    language: &5 !ruby/marshalable:Cucumber::Core::Ast::LanguageDelegator
      :__v2__:
      - :@iso_code
      ? - en
      : !ruby/object:Gherkin::Dialect
        spec:
          and:
          - "* "
          - 'And '
          background:
          - Background
          but:
          - "* "
          - 'But '
          examples:
          - Examples
          - Scenarios
          feature:
          - Feature
          - Business Need
          - Ability
          given:
          - "* "
          - 'Given '
          name: English
          native: English
          scenario:
          - Scenario
          scenarioOutline:
          - Scenario Outline
          - Scenario Template
          then:
          - "* "
          - 'Then '
          when:
          - "* "
          - 'When '
    location: &12 !ruby/struct:Cucumber::Core::Ast::Location::Precise
      file: features/example2.feature
      lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
        data: !ruby/object:Set
          hash:
            2: true
    background: !ruby/object:Cucumber::Core::Ast::EmptyBackground {}
    comments: []
    tags:
    - !ruby/object:Cucumber::Core::Ast::Tag
      location: &1 !ruby/struct:Cucumber::Core::Ast::Location::Precise
        file: features/example2.feature
        lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
          data: !ruby/object:Set
            hash:
              1: true
      name: "@example"
      all_locations:
      - *1
    - !ruby/object:Cucumber::Core::Ast::Tag
      location: &2 !ruby/struct:Cucumber::Core::Ast::Location::Precise
        file: features/example2.feature
        lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
          data: !ruby/object:Set
            hash:
              1: true
      name: "@p"
      all_locations:
      - *2
    keyword: Feature
    name: Example Feature
    description: |2-
        When I want to learn how to make test cases
        As a user of the test automation tool
        I want to run and adjust the tests below
    feature_elements:
    - !ruby/object:Cucumber::Core::Ast::Scenario
      location: &7 !ruby/struct:Cucumber::Core::Ast::Location::Precise
        file: features/example2.feature
        lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
          data: !ruby/object:Set
            hash:
              8: true
      comments: []
      tags:
      - !ruby/object:Cucumber::Core::Ast::Tag
        location: &3 !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                7: true
        name: "@example04"
        all_locations:
        - *3
      - !ruby/object:Cucumber::Core::Ast::Tag
        location: &4 !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                7: true
        name: "@a"
        all_locations:
        - *4
      keyword: Scenario
      name: example04 - Google Search
      description: ''
      raw_steps:
      - !ruby/object:Cucumber::Core::Ast::Step
        language: *5
        location: &6 !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                9: true
        comments: []
        keyword: 'Given '
        name: I navigate to Google in english
        multiline_arg: !ruby/object:Cucumber::Core::Ast::EmptyMultilineArgument {}
        all_locations:
        - *6
      all_locations:
      - *7
      - *3
      - *4
    - &13 !ruby/object:Cucumber::Core::Ast::Scenario
      location: &11 !ruby/struct:Cucumber::Core::Ast::Location::Precise
        file: features/example2.feature
        lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
          data: !ruby/object:Set
            hash:
              12: true
      comments: []
      tags:
      - !ruby/object:Cucumber::Core::Ast::Tag
        location: &8 !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                11: true
        name: "@example05"
        all_locations:
        - *8
      - !ruby/object:Cucumber::Core::Ast::Tag
        location: &9 !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                11: true
        name: "@a"
        all_locations:
        - *9
      keyword: Scenario
      name: example05 - Google Search
      description: ''
      raw_steps:
      - &16 !ruby/object:Cucumber::Core::Ast::Step
        language: *5
        location: &10 !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                13: true
        comments: []
        keyword: 'Given '
        name: I navigate to Google in english
        multiline_arg: !ruby/object:Cucumber::Core::Ast::EmptyMultilineArgument {}
        all_locations:
        - *10
      all_locations:
      - *11
      - *8
      - *9
    - !ruby/object:Cucumber::Core::Ast::Scenario
      location: !ruby/struct:Cucumber::Core::Ast::Location::Precise
        file: features/example2.feature
        lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
          data: !ruby/object:Set
            hash:
              16: true
      comments: []
      tags:
      - !ruby/object:Cucumber::Core::Ast::Tag
        location: !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                15: true
        name: "@example06"
      - !ruby/object:Cucumber::Core::Ast::Tag
        location: !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                15: true
        name: "@b"
      keyword: Scenario
      name: example06 - Google Search
      description: ''
      raw_steps:
      - !ruby/object:Cucumber::Core::Ast::Step
        language: *5
        location: !ruby/struct:Cucumber::Core::Ast::Location::Precise
          file: features/example2.feature
          lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
            data: !ruby/object:Set
              hash:
                17: true
        comments: []
        keyword: 'Given '
        name: I navigate to Google in english
        multiline_arg: !ruby/object:Cucumber::Core::Ast::EmptyMultilineArgument {}
    all_locations:
    - *12
    - *1
    - *2
  - *13
  - !ruby/object:Cucumber::Hooks::BeforeHook
    location: &14 !ruby/struct:Cucumber::Core::Ast::Location::Precise
      file: lapis_lazuli-0.8.7/lib/lapis_lazuli/cucumber.rb
      lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
        data: !ruby/object:Set
          hash:
            9: true
  action: !ruby/object:Cucumber::Core::Test::UnskippableAction
    location: *14
    block: !ruby/object:Proc {}
    timer: !ruby/object:Cucumber::Core::Test::Timer
      start_time: 1496148943246929
- !ruby/object:Cucumber::Core::Test::Step
  source:
  - *15
  - *13
  - *16
  action: !ruby/object:Cucumber::Core::Test::Action
    location: !ruby/struct:Cucumber::Core::Ast::Location::Precise
      file: features/step_definitions/interaction_steps.rb
      lines: !ruby/struct:Cucumber::Core::Ast::Location::Lines
        data: !ruby/object:Set
          hash:
            6: true
    block: !ruby/object:Proc {}
    timer: !ruby/object:Cucumber::Core::Test::Timer
      start_time: 1496148944286809
source:
- *15
- *13
around_hooks: []
```

**Comment by sjieg** [comment]
Was fixed in pull #44
